### ISSUE TITLE
Wiserhub V2 Api

### ISSUE THREAD

**Issue by msp1974** [asantaga/wiserheatingapi#14]
While investigating some of the other functions of the wiser app that are not yet supported in this api, I found that the wiser app uses a v2 api to the hub.

For reading the hub domain data this is significantly faster and it also separates schedule data into a different api call with different json format.

I suspect that a future hub firmware update may remove this v1 api in favor of v2, so we should have on the enhancement list to migrate to the v2 version.

As a side note, found a really great app for phone or tablet to monitor http calls from apps similar to fiddler called HTTPCanary. Was very useful to do the function in the app and see the calls to the api.

**Comment by asantaga** [comment]
Nice, 

Moving to v2 we should plan for, but I dont think they'll remove the v1 api for a long time.. unless they need to why would they?

**Comment by msp1974** [comment]
Agreed but thought worth logging issue so it's on radar.

You not busy today? ;-)

**Comment by asantaga** [comment]
I was manic on Tuesday... Just seeing what I need to do for tomorrow :-)

**Comment by keirstitt84** [comment]
> 
> 
> Nice,
> 
> Moving to v2 we should plan for, but I dont think they'll remove the v1 api for a long time.. unless they need to why would they?

If i was an engineer at schnieder then at some point I might run out of headroom on the Atmel they are using. Maintaining support for an old API is expensive so I wouldn't want this project to die on the assumption that they won't drop it.

That said the availability of this library was a key to why I purchased the Drayton system so hopefully they will work with us.

**Comment by keirstitt84** [comment]
> 
> 
> While investigating some of the other functions of the wiser app that are not yet supported in this api, I found that the wiser app uses a v2 api to the hub.
> 
> For reading the hub domain data this is significantly faster and it also separates schedule data into a different api call with different json format.
> 
> I suspect that a future hub firmware update may remove this v1 api in favor of v2, so we should have on the enhancement list to migrate to the v2 version.
> 
> As a side note, found a really great app for phone or tablet to monitor http calls from apps similar to fiddler called HTTPCanary. Was very useful to do the function in the app and see the calls to the api.

Do you think you could give a quick brief as to what you know about how to access this API? 

Just as a starting point.

**Comment by msp1974** [comment]
I did some work some months ago on updating this api to both use the v2 version and also to shift most of the work of managing the integration to the Wiser hub out of the HA component, to make it easier to get it into the core HA release.  Unfortunately, I've not yet had time to finish it but it is mostly there and mostly tested.  You can see in on my github repo at [Wiser API v2](https://github.com/msp1974/wiserHeatAPIv2).

**Comment by asantaga** [comment]
Yeah I was thinking about that ,  @msp1974 , lets chat offline about how to merge it all in..
